File: Python/Reverse Linked List.py
class Solution:
    # @param {ListNode} head
    # @return {ListNode}
    def reverseList(self, head):
        mockhead = ListNode(0)
        while head:
            p = head.next
            head.next = mockhead.next
            mockhead.next = head
            head = p
        return mockhead.next

File: java/Reverse_Linked_List.java
// Iteratively
// Time O(n), space O(1).

public class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode pre = null, p = head;
        while (p != null) {
            head = p.next;
            p.next = pre;
            pre = p;
            p = head;
        }
        return pre;
    }
}
